stackable-certs/src/keys/mod.rs

```diff
@@ -0,0 +1,66 @@
+//! Contains primitives to create private keys, which are used to sign CAs
+//! and bind to leaf certificates.
+//!
+//! This module currently provides the following algorithms:
+//!
+//! ## ECDSA
+//!
+//! In order to work with ECDSA keys, this crate requires two dependencies:
+//! [`ecdsa`], which provides primitives and traits, and [`p256`] which
+//! implements the NIST P-256 elliptic curve and supports ECDSA.
+//!
+//! ```
+//! use stackable_certs::sign::ecdsa::SigningKey;
+//! let key = SigningKey::new().unwrap();
+//! ```
+//!
+//! ## RSA
+//!
+//! In order to work with RSA keys, this crate requires the [`rsa`] dependency.
+//!
+//! ```
+//! use stackable_certs::sign::rsa::SigningKey;
+//! let key = SigningKey::new().unwrap();
+//! ```
+//!
+//! It should be noted, that the crate is currently vulnerable to the recently
+//! discovered Marvin attack. The `openssl` crate is also impacted by this. See:
+//!
+//! - <https://people.redhat.com/~hkario/marvin/>
+//! - <https://rustsec.org/advisories/RUSTSEC-2023-0071.html>
+//! - <https://github.com/RustCrypto/RSA/issues/19>
+
+use std::fmt::Debug;
+
+use p256::pkcs8::EncodePrivateKey;
+use signature::{Keypair, Signer};
+use x509_cert::spki::{EncodePublicKey, SignatureAlgorithmIdentifier, SignatureBitStringEncoding};
+
+pub mod ecdsa;
+pub mod rsa;
+
+// NOTE (@Techassi): This can _maybe_ be slightly simplified by adjusting the
+// trait and using a blanket impl on types which implement Deref<Target = _>.
+pub trait KeypairExt
```

Comment:
```suggestion
pub trait Keypair
```

This isn't an extension trait, it's semantically meaningful.

Reply:
I actually need to revert this change, because it clashes with `signature::Keypair` when both traits are required to be in scope.

Reply:
Do you have an idea for a better name, which doesn't clash with the external trait?

Reply:
`CertificateKeypair`.. maybe? But it also feels kind of redundant to end up needing both traits that do more or less the same thing.

Reply:
Renamed in c96fa62

There are a few ways to simplify the trait bounds. If we want to do that, I would do that in a separate PR.

Or we could get rid of it entirely, if we only support one type of key, either RSA or ECDSA.